terminus/models/building.py

```diff
@@ -1,20 +1,27 @@
 from city_model import CityModel
 
 class Building(CityModel):
-    def __init__(self, origin, name=None):
+    def __init__(self, origin, size, height, name=None):
+    """
+    Generates a square building of base (size * size) and the given height,
+    located at position 'origin'.
+    """
         super(Building, self).__init__(name)
         self.origin = origin
+        self.size = size
+        self.height = height
 
     def template(self):
         return """
           <model name="{{model.name}}">
              <static>1</static>
              <link name="{{model.name}}_link">
-                <pose frame="">{{model.origin.x}} {{model.origin.y}} {{model.origin.z + 20}} 0 0 0</pose>
+                <pose frame="">{{model.origin.x}} {{model.origin.y}}
+                    {{model.origin.z + model.height / 2}} 0 0 0</pose>
                 <collision name="{{model.name}}_collision">
```

Comment:
Very minor, but I would like to have all the computations in the object itself and leave to the templates only the rendering.